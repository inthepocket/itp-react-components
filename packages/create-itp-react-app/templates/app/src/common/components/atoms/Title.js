import React from 'react';
import { string } from 'prop-types';
import styles from './Title.module.css';

const Title = ({ text }) => <h1 className={styles.title}>{text}</h1>;

Title.propTypes = {
  text: string,
};

Title.defaultProps = {
  text: 'Default title',
};

export default Title;
