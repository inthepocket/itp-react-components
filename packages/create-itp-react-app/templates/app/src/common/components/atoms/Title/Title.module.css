.Title {
  font-family: "Flama Semicondensed", sans-serif;
  font-size: 4.5vw;
  font-weight: 400;
}
