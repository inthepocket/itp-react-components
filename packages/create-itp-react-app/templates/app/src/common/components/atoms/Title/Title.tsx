import * as React from 'react';
import { default as styles } from './Title.module.css';

const Title = ({
  text,
}: TitleProps) => <h1 className={styles.Title}>{text}</h1>;

interface TitleProps {
  text: string;
}

export default Title;
