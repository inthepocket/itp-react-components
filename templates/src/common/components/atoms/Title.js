import React from 'react';
import { string } from 'prop-types';
import './Title.css';

const Title = ({ text }) => (
  <h1 className="title">{text}</h1>
);

Title.propTypes = {
  text: string,
};

Title.defaultProps = {
  text: 'Default title',
};

export default Title;
