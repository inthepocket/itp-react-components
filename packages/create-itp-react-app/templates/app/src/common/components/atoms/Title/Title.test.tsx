import * as React from 'react';
import reactTestRenderer from 'react-test-renderer';
import Title from './Title';

test('Title snapshot: default', () => {
  const tree = reactTestRenderer
    .create((
      <Title text="foo:bar" />
    ))
    .toJSON();

  expect(tree).toMatchSnapshot();
});
