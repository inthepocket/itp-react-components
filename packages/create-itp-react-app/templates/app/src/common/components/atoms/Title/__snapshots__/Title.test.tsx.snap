// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title snapshot: default 1`] = `
<h1
  className="Title"
>
  foo:bar
</h1>
`;
